op-preimage/iface.go

```diff
@@ -62,6 +68,40 @@ func (k Keccak256Key) TerminalString() string {
 	return "0x" + hex.EncodeToString(k[:])
 }
 
+// Sha256Key wraps a sha256 hash to use it as a typed pre-image key.hash
+type Sha256Key [32]byte
+
+func (k Sha256Key) PreimageKey() (out [32]byte) {
+	out = k
+	out[0] = byte(Sha256KeyType)
+	return
+}
+
+func (k Sha256Key) String() string {
+	return "0x" + hex.EncodeToString(k[:])
```

Comment:
Can use go-ethereum's `hexutil` package to [`Encode(b []byte) string`](https://pkg.go.dev/github.com/ethereum/go-ethereum/common/hexutil#Encode) bytes with a string like so.
(just need to import `github.com/ethereum/go-ethereum/common/hexutil`).
Not a big deal so will leave it up to you if you prefer to be explicit.
```suggestion
	return hexutil.Encode(k[:])
```

Reply:
Go mod doesn't seem to like pulling this in; Will keep parity with the old format that the other keys use.

Reply:
Sounds good, nbd just pointing it out 👍

Reply:
Yeah `op-preimage` is deliberately very light on dependencies so I don't believe it has any reference to go-ethereum currently.